src/frontends/sql/decode_test.cc

```diff
@@ -202,9 +203,104 @@ const TextprotoDeathMessagePair kTextprotoDeathMessagePairs[] = {
      .death_message = "Required field id was not present in Expression."},
     {.textproto = R"(id: 1)",
      .death_message = "Required field expr_variant not set."},
+    {.textproto = R"(id: 1 merge_operation: { })",
+     .death_message = "Each MergeOperation is expected to have at least one "
+                      "non-control input."},
 };
 
 INSTANTIATE_TEST_SUITE_P(DecodeExprDeathTest, DecodeExprDeathTest,
                          ValuesIn(kTextprotoDeathMessagePairs));
 
+// We test the decoding of the child `Operation`s in other tests. We also
```

Comment:
I am sharing the test format `IdNameAndStringTest` between `DecodeSourceTableColumnExprTest` and `DecodeLiteralExprTest`. That said, that's not much code and could be split off into its own header. I think splitting these up could make sense, but I think it would increase the footprint of this PR significantly, so I'd rather do it as a follow-on PR if that's ok.

Reply:
SG, but let us get it done in a follow up?

Reply:
Absolutely, will do as my next PR.